### ISSUE TITLE
Clarify hyperparameter

### ISSUE THREAD

**Issue by chubzchubz97** [JinheonBaek/KALMV#4]
Hi,

Within the file (sh ./scripts/odqa_run.sh), what's the difference between the first and second line?

**Comment by JinheonBaek** [comment]
Thank you for your interest in our work. The first line indicates the setting without any rectification step; meanwhile, the second line indicates one rectification step with the additional parameters of "--edit_output True --num_edits 1".

**Comment by chubzchubz97** [comment]
Thank you for your time. Then, are the results in Table 1 from the second line (with edits)? Do you have the exact lines to reproduce the results for your method in Table 1?

Also, will you share the code of other methods on the datasets (at least for ODQA)?

**Comment by JinheonBaek** [comment]
Yes, the results in Table 1 are from the second line. Unfortunately, I no longer have the codes for other methods.

**Comment by chubzchubz97** [comment]
Hi, can I confirm that in Table 1, only the size of the LM is changed and the size of retriever is always base?

**Comment by JinheonBaek** [comment]
Yes, I can confirm it.

**Comment by chubzchubz97** [comment]
Thank you. I also have the following questions:
1. How can I save the best model after training?
2. Do you have the code to replicate the results in Table 4?
3. The eval script does not run any inference. Instead, it only prints and save summary results?

**Comment by chubzchubz97** [comment]
[Answer Generation Results] Original - Acc: 0.1703601108033241, F1: 0.19309443836525028, EM: 0.13905817174515236
[Answer Generation Results] Filtered - Acc: 0.5249088699878494, F1: 0.5472347518823827, EM: 0.44349939246658565
[Verification Results] Precision: 0.5249088699878494, Recall: 0.7024390243902439, F1: 0.6008344923504867
[Verification Results] All Acc: 0.7889196675900277, Retrieval Acc: 0.8168975069252078, Grounding Acc: 0.9199445983379502, Generation Acc: 0.8409972299168975

Hi, for the above, what's the difference between the third and fourth line? Also, what's generation acc and how is it different from acc in the second line?

**Comment by chubzchubz97** [comment]
Hi, and the following line seems to produce much lower performance compared to in table 1.. third line in odqa_run.sh

python main.py --data_name HotpotQA --data_type ODQA --knowledge_base wiki --model_type flan --model_name_or_path google/flan-t5-base --batch_size 32 --verifier_num_epochs 3 --verifier_batch_size 6 --verifier_sample True --ensemble True --edit_output True --num_edits 1 --exp_name KALMV_Rectify-1

Do you have the hyperparameter settings for large and xl sizes?

**Comment by JinheonBaek** [comment]
> Thank you. I also have the following questions:
> 
> 1. How can I save the best model after training?
> 2. Do you have the code to replicate the results in Table 4?
> 3. The eval script does not run any inference. Instead, it only prints and save summary results?

1. The model saving function is not implemented in our code. I kindly would like to suggest you to refer to torch.save docs (https://pytorch.org/tutorials/beginner/saving_loading_models.html). 
2. This is the only code that I have. To replicate the results in Table 4, you can simply adjust the training and test datasets.
3. Yes, the eval script calculates the results, from which printing and saving the results as well.

**Comment by JinheonBaek** [comment]
> [Answer Generation Results] Original - Acc: 0.1703601108033241, F1: 0.19309443836525028, EM: 0.13905817174515236 [Answer Generation Results] Filtered - Acc: 0.5249088699878494, F1: 0.5472347518823827, EM: 0.44349939246658565 [Verification Results] Precision: 0.5249088699878494, Recall: 0.7024390243902439, F1: 0.6008344923504867 [Verification Results] All Acc: 0.7889196675900277, Retrieval Acc: 0.8168975069252078, Grounding Acc: 0.9199445983379502, Generation Acc: 0.8409972299168975
> 
> Hi, for the above, what's the difference between the third and fourth line? Also, what's generation acc and how is it different from acc in the second line?

The third line corresponds to the results of Figure 3; meanwhile, the fourth line corresponds to the results of Figure 2. The generation acc indicates the ratio to which the verifier predicts the correct generation case correctly, which differs from the (filtered) answer generation accuracy (Table 1) from the generator model.

**Comment by JinheonBaek** [comment]
> Hi, and the following line seems to produce much lower performance compared to in table 1.. third line in odqa_run.sh
> 
> python main.py --data_name HotpotQA --data_type ODQA --knowledge_base wiki --model_type flan --model_name_or_path google/flan-t5-base --batch_size 32 --verifier_num_epochs 3 --verifier_batch_size 6 --verifier_sample True --ensemble True --edit_output True --num_edits 1 --exp_name KALMV_Rectify-1
> 
> Do you have the hyperparameter settings for large and xl sizes?

The provided hyperparameter in https://github.com/JinheonBaek/KALMV/blob/main/scripts/odqa_run.sh should be the one that we used to report the results in Table 1. Could you share the results? 

In addition, the hyperparameters for large and xl size models are the same; you can simply change the model_name_or_path variable (e.g., --model_name_or_path google/flan-t5-large).
